be/src/runtime/routine_load/data_consumer_pool.cpp

```diff
@@ -71,11 +71,12 @@ Status DataConsumerPool::get_consumer_grp(StreamLoadContext* ctx, std::shared_pt
     }
     DCHECK(ctx->kafka_info);
 
-    std::shared_ptr<KafkaDataConsumerGroup> grp = std::make_shared<KafkaDataConsumerGroup>();
-
     // one data consumer group contains at least one data consumers.
     int max_consumer_num = config::max_consumer_num_per_group;
     size_t consumer_num = std::min((size_t)max_consumer_num, ctx->kafka_info->begin_offset.size());
+
+    std::shared_ptr<KafkaDataConsumerGroup> grp = std::make_shared<KafkaDataConsumerGroup>(consumer_num);
```

Comment:
Will the consumer_num will be very large and the thread pool have much threads?

Reply:
By default the number of consumer is 3, but it can be set manually by user.